File: backend/dataset/rotate/rotate_17.py
class Solution:
    def rotate(self, a: List[int], k: int) -> None:
        n = len(a)
        c = 1
        if k != 0:
            if k != 1:
                c = gcd(n, k)
            for i in range(c):
                j = (i + k)%n
                b = a[i]
                while j != i:
                    d = a[j]
                    a[j] = b
                    b = d
                    j = (j+k)%n
                a[j] = b

    def gcd(m, n):
        if m % n == 0:
            return n
        else:
            gcd(n, m%n)

File: backend/CDataset/rotate/rotate_17.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        if(k%nums.size() && nums.size()>1 ){
            int t = nums.size()-k%nums.size();
            for(int i= 0;i<t;i++){
                nums.push_back(nums[i]);
            }
            nums.erase(nums.begin(),nums.begin() + t);
        }
    }
};
